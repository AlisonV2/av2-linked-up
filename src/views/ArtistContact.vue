<template>
  <div class="container">
    <form class="auth-form" @submit.prevent="handleSubmit">
      <div class="form-floating mb-3">
        <input
          type="text"
          class="form-control"
          placeholder="title"
          v-model="title"
          required
        />
        <label>Title</label>
      </div>
      <div class="form-floating mb-3">
        <input
          type="textarea"
          class="form-control"
          placeholder="Description"
          v-model="description"
          required
        />
        <label>Description</label>
      </div>
      <div class="btn-center">
        <app-button>Submit</app-button>
      </div>
    </form>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

export default {
  name: 'ArtistContact',
  data() {
    return {
      id: '',
      artist: {},
      userId: '',
      description: '',
      title: '',
    };
  },
  created() {
    this.$store.dispatch('getCurrentUser');
    this.id = this.$route.params.id;
    this.userId = this.$store.getters.getCurrentUser;
    this.artist = this.$route.params.artist;
  },
  methods: {
    handleSubmit() {
      const project = {
        clientId: this.userId,
        artistId: this.id,
        artistName: this.artist,
        title: this.title,
        description: this.description,
      };
      this.$store.dispatch('setNewProject', project).then(() => {
        this.$router
          .push({ name: 'AdminProjects' })
          .catch((err) => Sentry.captureException(err));
      });
    },
  },
};
</script>

<style></style>
