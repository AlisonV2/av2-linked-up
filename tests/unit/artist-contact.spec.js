import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import ArtistContact from '@/views/ArtistContact';
import AppButton from '@/components/app/Button';

const actions = {
  getCurrentUser: jest.fn(),
  setNewProject: jest.fn(),
};

const getters = {
  getCurrentUser() {
    return {
      uid: '1',
      displayName: 'User',
    };
  },
};

const store = createStore({
  actions,
  getters,
});

const $router = {
  push: jest.fn(),
};

const $route = {
  params: { id: '1', artist: 'Artist' },
};

const config = {
  data() {
    return {
      artist: {
        id: 1,
        name: 'Artist',
        description: 'Description',
        thumbnail: 'Thumbnail',
        socialLink: 'Social',
        shop: 'Shop',
        insta: 'Insta',
        city: 'City',
        zip: 'Zip',
        style: 'Style',
      },
      id: '1',
      userId: '1',
      description: 'Blablabla description',
      title: 'Blabla',
    };
  },
  global: {
    plugins: [store],
    stubs: {
      'app-button': AppButton,
      'router-link': RouterLinkStub
    },
    mocks: {
      $router,
      $route,
    },
  },
};

/**
 * @module ArtistContactTest
 */
describe('ArtistContact.vue', () => {
  it('ArtistContact Snapshot', () => {
    const component = shallowMount(ArtistContact, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getCurrentUser action should be called', async () => {
    const component = shallowMount(ArtistContact, config);
    ArtistContact.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getCurrentUser).toHaveBeenCalled;
  });

  it('getCurrentUser getter should be called', async () => {
    const component = shallowMount(ArtistContact, config);
    ArtistContact.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getCurrentUser).toHaveBeenCalled;
  });

  it('handleSubmit should be called with correct route params', async () => {
    const component = shallowMount(ArtistContact, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect($router.push).toHaveBeenCalled;
  });

  it('handleSubmit should trigger setNewProject action', async () => {
    const component = shallowMount(ArtistContact, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect(actions.setNewProject).toHaveBeenCalled;
  });
});
