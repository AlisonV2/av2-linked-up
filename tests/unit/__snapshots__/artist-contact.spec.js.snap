// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistContact.vue ArtistContact Snapshot 1`] = `
<div
  class="container"
>
  <form
    class="auth-form"
  >
    <div
      class="form-floating mb-3"
    >
      <input
        class="form-control"
        placeholder="title"
        required=""
        type="text"
      />
      <label>
        Title
      </label>
    </div>
    <div
      class="form-floating mb-3"
    >
      <input
        class="form-control"
        placeholder="Description"
        required=""
        type="textarea"
      />
      <label>
        Description
      </label>
    </div>
    <div
      class="btn-center"
    >
      <button
        class="btn"
      >
        
        Submit
        
      </button>
    </div>
  </form>
</div>
`;
